tracer/src/Datadog.Trace/Agent/IApiRequest.cs

```diff
@@ -20,5 +21,7 @@ internal interface IApiRequest
         Task<IApiResponse> PostAsync(ArraySegment<byte> bytes, string contentType, string contentEncoding);
 
         Task<IApiResponse> PostAsync(Func<Stream, Task> writeToRequestStream, string contentType, string contentEncoding, string multipartBoundary);
+
+        Task<IApiResponse> PostAsync(MultipartFormItem[] items, MultipartCompression multipartCompression = MultipartCompression.None);
```

Comment:
Thinking loud, can we refactor this interface to use extension methods, allowing the internal implementation to avoid having to implement all the methods?